### PR TITLE
t/op/coresubs.t:  Adapt to honor warnings-by-default

### PR DESCRIPTION
**Pull request by jkeenan**
Satisfies https://github.com/atoomic/perl/issues/102

### CODE REVIEW

File: t/op/coresubs.t [hunk 1]
```diff
@@ -139,9 +144,17 @@ sub inlinable_ok {
...
     my $my_code = $core_code =~ s/CORE::$word/my$word/r;
-    my $core = op_list(eval $core_code or die);
-    my $my   = op_list(eval   $my_code or die);
-    is $my, $core, "inlinability of CORE::$word $preposition parens $desc_suffix";
+    unless ($word eq 'push' or $word eq 'unshift') {
+        my $core = op_list(eval $core_code or die);
+        my $my   = op_list(eval   $my_code or die);
+        is $my, $core, "inlinability of CORE::$word $preposition parens $desc_suffix";
+    }
+    else {
```
Comment (atoomic) [inline t/op/coresubs.t:152]: much easier to read if move that condition being `if ( $word eq 'push' || $word eq 'unshift' )`

**Review by atoomic — COMMENTED** [review]
This LGTM, one minor comment which is more style/readability of the `unless`

File: t/op/coresubs.t (hunk 1 above)
Comment (jkeenan) [reply t/op/coresubs.t:152]: > much easier to read if move that condition being `if ( $word eq 'push' || $word eq 'unshift' )`

While I'm willing to do that, let me say why I wrote it that way.  The 'push' and 'unshift' cases will be much less frequent than all the others.  I believe one coding principle is to place the conditions that are going to be encountered more frequently at the top of an if-elsif-else block.  Does that not apply here?

Thank you very much.
Jim Keenan

Comment (jkeenan) [reply t/op/coresubs.t:152]: @atoomic Any further feedback?

thanks.
jimk

Comment (atoomic) [reply t/op/coresubs.t:152]: IMO any usage of unless outside `unless $condition` or `unless $x and $y` is hard to read and prone to errors.

As I said what you have here is not wrong, but will probably give some headache to most persons reading this line.
I personally recommend changing it, but this is just my 2cts.

Comment (jkeenan) [reply t/op/coresubs.t:152]: > IMO any usage of unless outside `unless $condition` or `unless $x and $y` is hard to read and prone to errors.
> 
> As I said what you have here is not wrong, but will probably give some headache to most persons reading this line.
> I personally recommend changing it, but this is just my 2cts.

Please review 406158614e4230fffc7d105ad6c9a804489dcbf9

Thanks.

Comment (jkeenan) [reply t/op/coresubs.t:152]: No response in 7 days.  Merged branch in commit cb074e0ca9.  Closing ticket.

Thank you very much.
Jim Keenan

**Pull request closed without merging**